plaso/storage/redis/redis_store.py

```diff
@@ -0,0 +1,489 @@
+
+  def _GetAttributeContainers(self, container_type):
+    """Yields attribute containers
+
+    Args:
+      container_type (str): container type attribute of the container being
+          added.
+
+    Yields:
+      AttributeContainer: attribute container.
+    """
+    container_key = self._GenerateRedisKey(container_type)
+    for identifier, serialized_data in self._redis_client.hscan_iter(
+        container_key):
+      attribute_container = self._DeserializeAttributeContainer(
+          container_type, serialized_data)
+
+      identifier_string = identifier.decode('utf-8')
+      redis_identifier = identifiers.RedisKeyIdentifier(identifier_string)
+      attribute_container.SetIdentifier(redis_identifier)
+      yield attribute_container
+
+  def _GetAttributeContainerByIdentifier(self, container_type, identifier):
+    """Retrieves the container with a specific identifier.
+
+    Args:
+      container_type (str): container type.
+      identifier (RedisKeyIdentifier): event data identifier.
+
+    Returns:
+      AttributeContainer: attribute container or None if not available.
+    """
+    container_key = self._GenerateRedisKey(container_type)
+    string_identifier = identifier.CopyToString()
+
+    serialized_data = self._redis_client.hget(
+        container_key, string_identifier)
+
+    if not serialized_data:
+      return None
+
+    attribute_container = self._DeserializeAttributeContainer(
+        container_type, serialized_data)
+
+    attribute_container.SetIdentifier(identifier)
+    return attribute_container
+
+  def _GetNumberOfAttributeContainers(self, container_type):
+    """Determines the number of containers of a type in the store.
+
+    Args:
+      container_type (str): attribute container type.
+
+    Returns:
+      int: the number of containers in the store of the specified type.
+    """
+    container_key = self._GenerateRedisKey(container_type)
+    return self._redis_client.hlen(container_key)
+
+  def _GetFinalizationKey(self):
+    """Generates the finalized key for the store.
+
+    Returns:
+      str: Redis key for the the finalization flag.
+    """
+    return '{0:s}-{1:s}'.format(
+        self._session_identifier, self._FINALIZED_KEY_NAME)
+
+  def _HasAttributeContainers(self, container_type):
+    """Determines if the store contains a specific type of attribute container.
+
+    Args:
+      container_type (str): attribute container type.
+
+    Returns:
+      bool: True if the store contains the specified type of attribute
+          containers.
+    """
+    container_key = self._GenerateRedisKey(container_type)
+    number_of_containers = self._redis_client.hlen(container_key)
+    return  number_of_containers > 0
+
+  def _RaiseIfNotReadable(self):
+    """Checks that the store is ready to for reading.
+
+     Raises:
+       IOError: if the store cannot be read from.
+       OSError: if the store cannot be read from.
+    """
+    if not self._redis_client:
+      raise IOError(
```

Comment:
the next two lines look like they can fit on a single line, please merge them.

Reply:
Done